src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/permissions/PermissionsClient.scala

```diff
@@ -1,98 +1,86 @@
 package org.broadinstitute.dsde.agora.server.dataaccess.permissions
 
 import AgoraPermissions._
-import org.broadinstitute.dsde.agora.server.AgoraConfig
+import org.broadinstitute.dsde.agora.server.dataaccess.{ReadAction, ReadWriteAction, WriteAction}
 import org.broadinstitute.dsde.agora.server.exceptions.PermissionNotFoundException
 import org.broadinstitute.dsde.agora.server.model.AgoraEntity
+import slick.jdbc.JdbcProfile
 
-import scala.concurrent.{Await, Future}
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration._
-import scala.util.Try
+import scala.util.{Failure, Success, Try}
 
-trait PermissionsClient {
-  import AgoraConfig.sqlDatabase.profile.api._
-  val db = AgoraConfig.sqlDatabase.db
+abstract class PermissionsClient(profile: JdbcProfile) {
+  import profile.api._
 
   val timeout = 10.seconds
 
   def alias(entity: AgoraEntity): String
 
+  def withPermissionNotFoundException[T](errorString: String = "Could not get permission")( op: => ReadWriteAction[T] ): ReadWriteAction[T] = {
+    op.asTry flatMap {
+      case Success(permission) => DBIO.successful(permission)
+      case Failure(ex) => DBIO.failed(new PermissionNotFoundException(errorString, ex))
+    }
+  }
+
   // Users
-  def addUserIfNotInDatabase(userEmail: String): Unit = {
+  def addUserIfNotInDatabase(userEmail: String): WriteAction[Int] = {
     // Attempts to add user to UserTable and ignores errors if user already exists
-    try {
-      Await.ready(db.run(users += UserDao(userEmail)), timeout)
-    } catch {
-      case _ : Throwable => //Do nothing
+    (users += UserDao(userEmail)).asTry flatMap {
+      case Success(count) => DBIO.successful(count)
+      case Failure(_) => DBIO.successful(0)
```

Comment:
yeah, that's a wider question about how we should track and fix little code annoyances, and i don't know the answer. normally i would say "fix them as you go", but i don't want the scope of this pr to creep until all of agora is clean and beautiful!

perhaps if there are a class of similar issues for them -- or a laundry list of things you happen to find -- they can all go in a ticket?